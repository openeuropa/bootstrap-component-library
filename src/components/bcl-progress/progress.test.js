import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import demoData from "@openeuropa/bcl-data-progress/data";

const template = "@oe-bcl/bcl-progress/progress.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Progress", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test(`renders correctly with striped style`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, striped: true })
    ).resolves.toMatchSnapshot();
  });

  test(`renders correctly with striped and animated style`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, striped: true, animated: true })
    ).resolves.toMatchSnapshot();
  });

  test(`renders correctly with variant danger`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, variant: "danger" })
    ).resolves.toMatchSnapshot();
  });

  test(`renders correctly with hidden label`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, hidden_label: true })
    ).resolves.toMatchSnapshot();
  });

  test(`renders correctly with message`, () => {
    expect.assertions(1);

    return expect(
      render({
        ...demoData,
        message: "This is a demo message under the progress bar.",
      })
    ).resolves.toMatchSnapshot();
  });

  test(`renders correctly with bar label`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, bar_label: "Loading..." })
    ).resolves.toMatchSnapshot();
  });
});
