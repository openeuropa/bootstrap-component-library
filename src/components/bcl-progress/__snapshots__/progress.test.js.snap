// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Progress renders correctly 1`] = `
<jest>
  <label
    class="progress-bar-label"
  >
    This is the label of progress bar
  </label>
  <div
    class="progress"
  >
    <div
      aria-label="25"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="25"
      class="progress-bar"
      role="progressbar"
      style="width: 25%"
    >
      25%
    </div>
  </div>
</jest>
`;

exports[`OE - Progress renders correctly with bar label 1`] = `
<jest>
  <label
    class="progress-bar-label"
  >
    This is the label of progress bar
  </label>
  <div
    class="progress"
  >
    <div
      aria-label="25"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="25"
      class="progress-bar"
      role="progressbar"
      style="width: 25%"
    >
      Loading...
    </div>
  </div>
</jest>
`;

exports[`OE - Progress renders correctly with hidden label 1`] = `
<jest>
  <label
    class="progress-bar-label"
  >
    This is the label of progress bar
  </label>
  <div
    class="progress"
  >
    <div
      aria-label="25"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="25"
      class="progress-bar"
      role="progressbar"
      style="width: 25%"
    />
  </div>
</jest>
`;

exports[`OE - Progress renders correctly with message 1`] = `
<jest>
  <label
    class="progress-bar-label"
  >
    This is the label of progress bar
  </label>
  <div
    class="progress"
  >
    <div
      aria-label="25"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="25"
      class="progress-bar"
      role="progressbar"
      style="width: 25%"
    >
      25%
    </div>
  </div>
  <small
    class="form-text text-muted progress-bar-message"
  >
    This is a demo message under the progress bar.
  </small>
</jest>
`;

exports[`OE - Progress renders correctly with striped and animated style 1`] = `
<jest>
  <label
    class="progress-bar-label"
  >
    This is the label of progress bar
  </label>
  <div
    class="progress"
  >
    <div
      aria-label="25"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="25"
      class="progress-bar progress-bar-striped progress-bar-animated"
      role="progressbar"
      style="width: 25%"
    >
      25%
    </div>
  </div>
</jest>
`;

exports[`OE - Progress renders correctly with striped style 1`] = `
<jest>
  <label
    class="progress-bar-label"
  >
    This is the label of progress bar
  </label>
  <div
    class="progress"
  >
    <div
      aria-label="25"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="25"
      class="progress-bar progress-bar-striped"
      role="progressbar"
      style="width: 25%"
    >
      25%
    </div>
  </div>
</jest>
`;

exports[`OE - Progress renders correctly with variant danger 1`] = `
<jest>
  <label
    class="progress-bar-label"
  >
    This is the label of progress bar
  </label>
  <div
    class="progress"
  >
    <div
      aria-label="25"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="25"
      class="progress-bar bg-danger"
      role="progressbar"
      style="width: 25%"
    >
      25%
    </div>
  </div>
</jest>
`;
